File: java/patternsample/src/main/java/pattern/sample/patternuse01/BattleGround.java
package pattern.sample.patternuse01;

import pattern.sample.patternuse01.building.Barrack;
import pattern.sample.patternuse01.code.UnitType;
import pattern.sample.patternuse01.unit.inf.Unit;

public class BattleGround {

    public static void main(String[] args) {

        Unit marine = Barrack.createUnit(UnitType.MARINE);

        marine.checkStatus();

        Unit medic = Barrack.createUnit(UnitType.MEDIC);

        medic.checkStatus();

    }
}

File: kotlin_basic/src/main/kotlin/bong/lines/patterns/abstractfactory/BattleGround.kt
package bong.lines.patterns.abstractfactory

import bong.lines.patterns.abstractfactory.building.Barrack
import bong.lines.patterns.abstractfactory.code.UnitType
import bong.lines.patterns.abstractfactory.unit.inf.Unit

fun main() {
    var marine: Unit = Barrack().create(UnitType.MARINE)

    marine.checkStatus()

    println()

    var medic: Unit = Barrack().create(UnitType.MEDIC)

    medic.checkStatus()
}